examples/Python/PET/interactive/spdhg_reconstruction.py

```diff
@@ -0,0 +1,310 @@
+    noisy_data = pet.AcquisitionData(sino_file)
+    randoms = pet.AcquisitionData(rand_file)
+    asm_norm = pet.AcquisitionSensitivityModel(norm_file)
+
+    am = pet.AcquisitionModelUsingRayTracingMatrix()
+    am.set_num_tangential_LORs(10)
+    # add it to the acquisition model
+    am.set_acquisition_sensitivity(asm_norm)
+    # Attenuation image
+    attn_image = pet.ImageData(attn_file)
+    asm_attn = pet.AcquisitionSensitivityModel(attn_image, am)
+    # converting attenuation into attenuation factors (see previous exercise)
+    asm_attn.set_up(noisy_data)
+    attn_factors = pet.AcquisitionData(noisy_data)
+    attn_factors.fill(1.0)
+    print('applying attenuation (please wait, may take a while)...')
+    asm_attn.unnormalise(attn_factors)
+    print('done')
+    # use these in the final attenuation model
+    asm_attn = pet.AcquisitionSensitivityModel(attn_factors)
+    # chain attenuation and normalisation
+    asm = pet.AcquisitionSensitivityModel(asm_norm, asm_attn)
+    # update the acquisition model etc
+    am.set_acquisition_sensitivity(asm)
+
+    am.set_background_term(randoms)
+    # create initial image estimate of dimensions and voxel sizes
+    # compatible with the scanner geometry (included in the AcquisitionData
+    # object acq_data) and initialize each voxel to 1.0
+    nxny = (127, 127)
+    initial_image = noisy_data.create_uniform_image(1.0, nxny)
+    am.set_up(noisy_data, initial_image)
+    image = initial_image
+    background = randoms
+
+
+g_noreg = ZeroFun()
+
+
+g_reg = FGP_TV_SIRF(lambdaReg=.3,
+                    iterationsTV=1000,
+                    tolerance=1e-5,
+                    methodTV=0,
+                    nonnegativity=1,
+                    printing=0,
+                    device='cpu')
+#%%
+
+g_reg = TGV_TV_SIRF(regularisation_parameter=0.0005,
+                    alpha1=1,
+                    alpha0=0.7,
+                    iterations=250,
+                    LipshitzConst=12,
+                    device='cpu')
+
+'''
+g_reg = ROF_LLT_SIRF(regularisation_parameterROF=0.04,
+                     regularisation_parameterLLT=0.01,
+                     iterations=500,
+                     time_marching_parameter=0.00002,
+                     device='cpu')
+
+
+
+
+g_reg = Diff4th_SIRF(regularisation_parameter=3.5,
+                     edge_parameter=0.02,
+                     iterations=500,
+                     time_marching_parameter=0.001,
+                     device='cpu')
+'''
+#g_reg = IndicatorBox(lower=0,upper=1)
+#%%
+
+
+A = SubsetOperator(am.get_linear_acquisition_model(), 14)
+A_norms = [PowerMethodNonsquare(Ai, 10, x0=image.copy()) for Ai in A]
+#%%
+# increase the norms to allow for inaccuracies in their computation
+Ls = [1.05 * L for L in A_norms]
+
+f = [KullbackLeibler(op.sirf2sub(noisy_data), op.sirf2sub(background))
+     for op in A]
+
+#%%
+recon_noreg = spdhg(f, g_noreg, A, A_norms=Ls)
+
+#%%
+
+# set the number of iterations
+niter = 5
```

Comment:
As I commented via email, 5 "iterations" are not enough. Perhaps it is best to change this to:

nepochs = 5
nsubsets = 14 (see line 236)
niter = nsubsets * nepochs